lib/twilioconnection.js

```diff
@@ -386,6 +482,17 @@ class TwilioConnection extends StateMachine {
     sendOrEnqueue(message);
   }
 
+  /**
+   * Start the TCMP handshake.
+   * @private
+   */
+  _startHandshake() {
+    this.transition('connecting');
+    this._sendHello();
+    const { welcomeTimeout } = this._options;
+    this._welcomeTimeout = new Timeout(() => this._handleWelcomeTimeout(), welcomeTimeout);
```

Comment:
This may be covered in @makarandp0's other PR, but the welcome timeout should be 15 seconds not 5 seconds, and the welcome should be sent at 0, 5, and 10 seconds during this time.